streams/src/test/java/org/apache/kafka/streams/internals/ApiUtilsTest.java

```diff
@@ -0,0 +1,143 @@
+        } catch (final IllegalArgumentException e) {
+            assertEquals(VALIDATE_MILLISECOND_NULL_SUFFIX, e.getMessage());
+        }
+
+        // Null check with prefix
+        final String nullDurationPrefix = prepareMillisCheckFailMsgPrefix(null, "nullDuration");
+        try {
+            validateMillisecondDuration(null, nullDurationPrefix);
+        } catch (final IllegalArgumentException e) {
+            assertEquals(nullDurationPrefix + VALIDATE_MILLISECOND_NULL_SUFFIX, e.getMessage());
+        }
+
+        // Maximum Duration which is out of long range with out prefix
+        try {
+            ApiUtils.validateMillisecondDuration(Duration.ofDays(MAX_ACCEPTABLE_DAYS_FOR_DURATION), "");
+        } catch (final IllegalArgumentException e) {
+            assertEquals(VALIDATE_MILLISECOND_OVERFLOW_SUFFIX, e.getMessage());
+        }
+
+        // Maximum Duration which is out of long range with prefix
+        final String maxDurationPrefix = prepareMillisCheckFailMsgPrefix(Instant.MAX, "maxDuration");
+        try {
+            ApiUtils.validateMillisecondDuration(Duration.ofDays(MAX_ACCEPTABLE_DAYS_FOR_DURATION), maxDurationPrefix);
+        } catch (final IllegalArgumentException e) {
+            assertEquals(maxDurationPrefix + VALIDATE_MILLISECOND_OVERFLOW_SUFFIX, e.getMessage());
+        }
+    }
+
+    @Test
+    public void testValidateMillisecondInstantWithInvalidInput() {
+        // Null check wo prefix
+        try {
+            validateMillisecondInstant(null, "");
+        } catch (final IllegalArgumentException e) {
+            assertEquals(VALIDATE_MILLISECOND_NULL_SUFFIX, e.getMessage());
+        }
+
+        // Null check with prefix
+        final String nullInstantPrefix = prepareMillisCheckFailMsgPrefix(null, "nullInstant");
+        try {
+            validateMillisecondInstant(null, nullInstantPrefix);
+        } catch (final IllegalArgumentException e) {
+            assertEquals(nullInstantPrefix + VALIDATE_MILLISECOND_NULL_SUFFIX, e.getMessage());
+        }
+
+        // Maximum Instant which is out of long range  with out prefix
+        try {
+            validateMillisecondInstant(Instant.MAX, "");
+        } catch (final IllegalArgumentException e) {
+            assertEquals(VALIDATE_MILLISECOND_OVERFLOW_SUFFIX, e.getMessage());
+        }
+
+        // Maximum Instant which is out of long range  with prefix
+        final String maxInstantPrefix = prepareMillisCheckFailMsgPrefix(Instant.MAX, "maxInstant");
+        try {
+            validateMillisecondInstant(Instant.MAX, maxInstantPrefix);
+        } catch (final IllegalArgumentException e) {
+            assertEquals(maxInstantPrefix + VALIDATE_MILLISECOND_OVERFLOW_SUFFIX, e.getMessage());
+        }
+    }
+
+    // Happy path
+    @Test
+    public void testPrepareMillisCheckFailMsgPrefix() {
+        assertEquals(format(MILLISECOND_VALIDATION_FAIL_MSG_FRMT, "sampleInstant", sampleInstant),
```

Comment:
I think, we should provide the expected String in a hardcode way instead of using `format()`. Similar below.

Reply:
I have given thought on not to duplicate the functionality in `Test` classes as suggestion. then I  implemented check with `org.hamcrest.CoreMatchers#containsString()` on returned `message`. 

Ex:

    @Test
    public void shouldContainsNameAndValueInFailMsgPrefix() {
        final String failMsgPrefix = prepareMillisCheckFailMsgPrefix("someValue", "variableName");

        assertThat(failMsgPrefix, containsString("variableName"));
        assertThat(failMsgPrefix, containsString("someValue"));
    }


Do you think, we should have exact check with hard coded string?

Reply:
Depends how strict the test should be. I guess, I am fine both ways, with a slight preference for the IMHO simpler (and stricter) variant, to just user `assertEquals("expected string", methodCall());`

Reply:
Thanks for response during weekend 👍. I am fine with both options too. So, we have two options here,

1. strict check - more accurate and need to be changes for small corrections.
2. moderate check - moderately accurate(not 100%), no need to change as long as variable exists in any order place in message - little bit fan of it due to cosmetic change convenience for next commits.

Please let me know final choice on it. I'll make the code changes if needed. TA

Reply:
It's your PR. Your choice. :)

Reply:
At this time, I am thinking of settling with moderate check unless some one recommends strict check.